filebeat/fileset/config.go

```diff
@@ -0,0 +1,17 @@
+package fileset
+
+// ModuleConfig contains the configuration file options for a module
+type ModuleConfig struct {
+	Module   string                    `config:"module"     validate:"required"`
+	Enabled  *bool                     `config:"enabled"`
+	Filesets map[string]*FilesetConfig `config:"filesets"`
```

Comment:
The command lines don't include `filesets`, just the configuration. For the cli, the options are prefixed with `module.fileset`, so for example `nginx.access`. From there on, the CLI flags match the structure in the configuration file. So to set `close_eof` you would type `nginx.access.prospector.close_eof`. But anyway, I'll look into getting rid of `filesets`, that would be cleaner.

There are more things to a fileset config than the prospector. Each fileset has configurable variables, which can be used in the prospector template and for deciding the pipeline. For example, you can set `format: json` and that variable is used in all the places we need it. I'd expect the module variables to be what 90% users need, and changing something directly in the prospector is "expert" mode. The pipeline itself is also a configuration at the fileset level.